akka-docs/src/test/scala/docs/stream/operators/sourceorflow/MapAsyncs.scala

```diff
@@ -0,0 +1,123 @@
+package docs.stream.operators.sourceorflow
+
+import java.util.concurrent.TimeUnit
+
+import akka.NotUsed
+import akka.actor.typed.ActorRef
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.scaladsl.Behaviors
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.Timeout
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+/**
+ *
+ */
+object CommonMapAsync {
+  case class Event(sequenceNumber: Int)
+  case class EventProcessingRequest(evt: Event, replyTo: ActorRef[Int])
+
+  val guardian = Behaviors.receiveMessage[EventProcessingRequest] {
+    case EventProcessingRequest(evt, replyTo) =>
+      TimeUnit.MILLISECONDS.sleep(500)
```

Comment:
I don't think we want this (even if it doesn't make it into the documentation)

Reply:
Yeah, I wasn't too happy about using a `sleep`. I needed a costly operation. Just read your recommendation to use `after` in the other comment and it'll simplify the samples a lot. Thanks.